RHDEVDOCS-2335: Added a procedure to monitor application and image vulnerabilities metrics (#48153)
openshift/openshift-docs: modules/odc-monitoring-your-app-vulnerabilities.adoc

:_content-type: PROCEDURE
[id="odc-monitoring-your-application-image-vulnerabilities-metrics_{context}"]
= Monitoring your application and image vulnerabilities metrics

After you create applications in your project and deploy them, use the *Developer* perspective in the web console to see the metrics for your application dependency vulnerabilities across your cluster. The metrics help you to analyze the following image vulnerabilities in detail:

* Total count of vulnerable images in a selected project
* Severity-based counts of all vulnerable images in a selected project
* Drilldown into severity to obtain the details, such as count of vulnerabilities, count of fixable vulnerabilities, and number of affected pods for each vulnerable image

.Prerequisites
* You have installed the Red Hat Quay Container Security operator from the Operator Hub.
+
[NOTE]
====
The Red Hat Quay Container Security operator detects the vulnerabilities by scanning the images that are in the quay registry.
====

.Procedure

. For a general overview of the image vulnerabilities, on the navigation panel of the *Developer* perspective, click *Project* to see the project dashboard.

> . Click *Image Vulnerabilities* in the *Status* section. A pop-up opens containing the image vulnerabilities breakdown details, such as *Vulnerable Container Images* and *Fixable Container Images*.

Reviewer: ```suggestion
. Click *Image Vulnerabilities* in the *Status* section. The window that opens displays details such as *Vulnerable Container Images* and *Fixable Container Images*.
```
OR
```suggestion
. Click *Image Vulnerabilities* in the *Status* section to see details such as *Vulnerable Container Images* and *Fixable Container Images*.
```

- Use "window" instead of "pop-up."
- Apply minimalism.

. For a detailed vulnerabilities overview, click the *Vulnerabilities* tab on the project dashboard.

> .. To get more detail about the vulnerabilities of an image, click *Image Name* of that particular image.

Reviewer: ```suggestion
.. To get more detail about an image, click its name.
```
Minimalism.  The context on line 29 establishes that this is about vulnerabilities.

+
[NOTE]
====
You can see the total number of vulnerabilities in the *Total* column of all the listed images.

> ====

Reviewer: Minimalism. Does this note add value? This feature of the page is probably self-evident to users.

Author: Removed this note.

> .. View the default graph with all the vulnerabilities in the *Details* tab. To view the vulnerabilities of specific types in a graph format, click the *App dependency* or *Base image* toggle button. For example, if you click *App dependency*, you see the graph containing vulnerabilities specific to application dependency.

Reviewer: Edit for minimalism?

Author: Done

* Drilldown into severity to obtain the details, such as count of vulnerabilities, count of fixable vulnerabilities, and number of affected pods for each vulnerable image

.Prerequisites
* You have installed the Container Security operator from the Operator Hub.

.Procedure

. For a general overview of the image vulnerabilities, on the navigation panel of the *Developer* perspective, click *Project* to see the project dashboard.

. Click *Image Vulnerabilities* in the *Status* section. A pop-up opens containing the image vulnerabilities breakdown details, such as *Vulnerable Container Images* and *Fixable Container Images*.
+
[NOTE]
====
The different color icons implies the severity level of vulnerabilities. Red for Critical, orange for Medium, and yellow for Low.

.Viewing image vulnerabilities
image::odc_image_vulnerabilities.png[]
====

. For detailed vulnerabilities overview, click the *Vulnerabilities* tab on the project dashboard.

.. Select an *Image Name* for further drilldown to vulnerabilities details associated with that particular image.
+
[NOTE]
====
You can see the total number of vulnerabilities in the *Total* column of all the listed images.
====

> .. Apply filter on the vulnerabilities based on their *Severity*, *Package*, *Type*, *Source*, *Current Version*, and *Fixed in Version*.

Reviewer 2: We have filters based on `Type` and `Severity`.

Users can sort the Vulnerabilities table based on `Severity`, `Package`, `Type`, `Source`, `Current Version`, and `Fixed in Version`.

<img width="417" alt="image" src="https://user-images.githubusercontent.com/2561818/180730497-d953b0a0-91a5-41e5-8774-2fadb2abd59f.png">

.. Click a *Vulnerability* for drilldown to its details:

... When you click a vulnerability of type *Base image*, you see more information about that vulnerability from Red Hat Security Advisory (RHSA).

> ... When you click a vulnerability of type *App dependency*, you see more information about that vulnerability from the snyk security application.

Author: I maybe wrong: The top part displays vulnerability information in a graph format, but if we look at this particular step wherein you are going to click a particular vulnerability, you do not get to see the information in graph format, instead you get to see more detailed info about that vulnerability, So, no need to shuffle these points above.

![image](https://user-images.githubusercontent.com/94683525/180740022-6a518569-3bbf-4820-9ac1-f0ca99a029f6.png)

Also to get the readers know abt this graph feature, i added a separate step:

View the default graph with all the vulnerabilities in the *Details* tab. If you want to view the vulnerabilities of specific types in a graph format, click the *App dependency* or *Base image* toggle button. For example, if you click *App dependency*, you see the graph containing vulnerabilities specific to application dependency.

What you say?

> ... When you click a vulnerability of type *App dependency*, you see more information about that vulnerability from the snyk security application.

Reviewer 2: Thank you for adding one more step. Now, it makes sense.

= Monitoring your application and image vulnerabilities metrics

After you create applications in your project and deploy them, use the *Developer* perspective in the web console to see the metrics for your application dependency vulnerabilities across your cluster. The metrics help you to analyze the following image vulnerabilities in detail:

* Total count of vulnerable images in a selected project
* Severity-based counts of all vulnerable images in a selected project
* Drilldown into severity to obtain the details, such as count of vulnerabilities, count of fixable vulnerabilities, and number of affected pods for each vulnerable image

.Prerequisites
* You have installed the Red Hat Quay Container Security operator from the Operator Hub.

.Procedure

. For a general overview of the image vulnerabilities, on the navigation panel of the *Developer* perspective, click *Project* to see the project dashboard.

. Click *Image Vulnerabilities* in the *Status* section. A pop-up opens containing the image vulnerabilities breakdown details, such as *Vulnerable Container Images* and *Fixable Container Images*.
+
[NOTE]
====

> The different color icons implies the severity level of vulnerabilities. Red for Critical, orange for Medium, and yellow for Low.

Reviewer: ```suggestion
The colors of the icons imply the severity of vulnerability: red for High, orange for Medium, and yellow for Low.
```

Author: Moved the details of this note into a new conceptual module

.Viewing image vulnerabilities
image::odc_image_vulnerabilities.png[]
====

. For a detailed vulnerabilities overview, click the *Vulnerabilities* tab on the project dashboard.

> .. Select an *Image Name* for further drilldown to vulnerabilities details associated with that particular image.

Reviewer: IBM SG: drilldown adjective, noun
Do not use as a noun or adjective. Use "drill down" (two words) only as a verb in the context of business intelligence or if its use is already established for a product.
```suggestion
.. To get more detail about the vulnerabilities of an image, click the name of a particular image.
```

+
[NOTE]
====
You can see the total number of vulnerabilities in the *Total* column of all the listed images.
====

> .. View the default graph with all the vulnerabilities in the *Details* tab. If you want to view the vulnerabilities of specific types in a graph format, click the *App dependency* or *Base image* toggle button. For example, if you click *App dependency*, you see the graph containing vulnerabilities specific to application dependency.

Reviewer: ```suggestion
.. View the default graph with all the vulnerabilities in the *Details* tab. To view the vulnerabilities of specific types in a graph format, click the *App dependency* or *Base image* toggle button. For example, if you click *App dependency*, you see the graph containing vulnerabilities specific to application dependency.
```

> .. Optional: To customize the view of vulnerabilities details, apply a filter on the vulnerabilities based on their *Severity* and *Type* or sort the vulnerabilities based on their *Severity*, *Package*, *Type* *Source*, *Current Version*, and *Fixed in Version*.

Reviewer 3: ```suggestion
.. Optional: To customize the view of vulnerabilities details, apply a filter on the vulnerabilities based on their *Severity* and *Type* or sort the vulnerabilities based on their *Severity*, *Package*, *Type*, *Source*, *Current Version*, and *Fixed in Version*.
```

> .. Optional: To customize the view of vulnerabilities details, apply a filter on the vulnerabilities based on their *Severity* and *Type* or sort the vulnerabilities based on their *Severity*, *Package*, *Type*, *Source*, *Current Version*, and *Fixed in Version*.

Reviewer: ```suggestion
.. Optional: You can filter the list of vulnerabilities based on their *Severity* and *Type* or sort them by *Severity*, *Package*, *Type*, *Source*, *Current Version*, and *Fixed in Version*.
```

> .. Click a *Vulnerability* for drilldown to its details:

Reviewer: ```suggestion
.. To get more details, click any *Vulnerability*:
```

> 
> ... When you click a vulnerability of type *Base image*, you see more information about that vulnerability from Red Hat Security Advisory (RHSA).
> ... When you click a vulnerability of type *App dependency*, you see more information about that vulnerability from the snyk security application.

Reviewer: ```suggestion
+
* *Base image* vulnerabilities display information from a Red Hat Security Advisory (RHSA).
* *App dependency* vulnerabilities display information from the Snyk security application.
```
